### PR TITLE
Lecture 20: Reinforcement Learning & Control Through Inference in GM.

### PR DESCRIPTION
**Pull request by siddancha**
Rendered version: https://siddancha.github.io/pgm-spring-2019/notes/lecture-20/

### CODE REVIEW

File: _posts/2019-04-01-lecture-20.md [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+On the other hand, the inference-based policy produces the following probability of the trajectory $\tau$:
+
+$$
+\hat{p}(\tau) \propto \mathbf{1}[p(\tau) \neq 0] \prod_{t=1}^T \pi(\mathbf{a}_t | \mathbf{s}_t).
+$$
+
+Computing the KL divergence between these two distributions, we have
+
+$$
+-D_{KL}(\hat{p}(\tau) || p(\tau)) = \mathbb{E}_{\tau \sim \hat{p}(\tau)} \left[ \sum_{t=1}^T r(\mathbf{s}_t, \mathbf{a}_t) - \log \pi(\mathbf{a}_t | \mathbf{s}_t) \right] = \sum_{t=1}^T \mathbb{E}_{(\mathbf{s}_t, \mathbf{a}_t) \sim \hat{p}(\mathbf{s}_t, \mathbf{a}_t)}[r(\mathbf{s}_t, \mathbf{a}_t))] + \mathbb{E}_{\mathbf{s}_t \sim \hat{p}(\mathbf{s}_t) }[H(\pi(\mathbf{a}_t | \mathbf{s}_t))]
```
Comment (zhisbug) [inline _posts/2019-04-01-lecture-20.md]: make this equation spanned in two lines.

File: _posts/2019-04-01-lecture-20.md [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
+In the case for deterministic dynamics, we can express the probability of an optimal trajectory $\tau$ as
+
+$$
+p(\tau) \propto \left[ p(\mathbf{s}_1) \prod_{t=1}^T p(\mathbf{s}_{t+1} | \mathbf{s}_t \mathbf{a}_t) \right] \exp \left( \sum_{t=1}^T r(\mathbf{s}_t, \mathbf{a}_t) \right).
+$$
+
+On the other hand, the inference-based policy produces the following probability of the trajectory $\tau$:
+
+$$
+\hat{p}(\tau) \propto \mathbf{1}[p(\tau) \neq 0] \prod_{t=1}^T \pi(\mathbf{a}_t | \mathbf{s}_t).
```
Comment (zhisbug) [inline _posts/2019-04-01-lecture-20.md:268]: there is an extra slash here?

**Review by zhisbug — CHANGES_REQUESTED** [review]
Very good note overall. Please fix my minor comments and I will take a second look and then merge.

File: _posts/2019-04-01-lecture-20.md (hunk 1 above)
Comment (siddancha) [reply _posts/2019-04-01-lecture-20.md]: Resolved.

File: _posts/2019-04-01-lecture-20.md (hunk 2 above)
Comment (siddancha) [reply _posts/2019-04-01-lecture-20.md:268]: Sorry I don't think I understand what you mean. Could you please elaborate?

Comment (alshedivat) [reply _posts/2019-04-01-lecture-20.md:268]: Hmm, I don't understand either.

**Comment by alshedivat** [comment]
@siddancha, minor comment: please change editor's name to Hao.

@zhisbug, the notes look good to me. Shall we merger after the change above?

File: _posts/2019-04-01-lecture-20.md (hunk 2 above)
Comment (zhisbug) [reply _posts/2019-04-01-lecture-20.md:268]: I meant the $\neq$ symbol was not properly rendered here?

Comment (siddancha) [reply _posts/2019-04-01-lecture-20.md:268]: I double checked and the $\neq$ symbol looks okay to me.

Comment (siddancha) [reply _posts/2019-04-01-lecture-20.md]: > @siddancha, minor comment: please change editor's name to Hao.
> 
> @zhisbug, the notes look good to me. Shall we merger after the change above?

Changed editor's name to Hao.

**Comment by siddancha** [comment]
> @siddancha, minor comment: please change editor's name to Hao.
> 
> @zhisbug, the notes look good to me. Shall we merger after the change above?

Changed editor's name to Hao.

**Pull request merged**